misk-redis/src/main/kotlin/misk/redis/RealPipelinedRedis.kt

```diff
@@ -0,0 +1,518 @@
+package misk.redis
+
+import okio.ByteString
+import okio.ByteString.Companion.toByteString
+import redis.clients.jedis.args.ListDirection
+import redis.clients.jedis.params.SetParams
+import redis.clients.jedis.params.ZRangeParams
+import redis.clients.jedis.resps.Tuple
+import java.time.Duration
+import java.util.function.Supplier
+
+internal class RealPipelinedRedis(private val pipeline: JedisPipeline) : DeferredRedis {
+  override fun del(key: String): Supplier<Boolean> {
+    val keyBytes = key.toByteArray(charset)
+    val response = pipeline.del(keyBytes)
+    return Supplier { response.get() == 1L }
```

Comment:
Note to self: pipeline response object are _actually_ suppliers. I don't think we can get around wrapping them, however, since we often don't use the raw response types.